packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -438,417 +410,396 @@ function formatAvailabilitySnapshot(data: {
 
-        if (originalNonFixedHost) {
-          users = [...fixedHosts, originalNonFixedHost];
-        } else {
-          const attendeeEmailSet = new Set(booking.attendees.map((attendee) => attendee.email));
+          if (originalNonFixedHost) {
+            users = [...fixedHosts, originalNonFixedHost];
+          } else {
+            const attendeeEmailSet = new Set(booking.attendees.map((attendee) => attendee.email));
 
-          // In this case, the first booking user is a fixed host, so the chosen non-fixed host is added as an attendee of the booking
-          const nonFixedAttendeeHost = users.find(
-            (user) => !user.isFixed && attendeeEmailSet.has(user.email)
-          );
-          users = [...fixedHosts, ...(nonFixedAttendeeHost ? [nonFixedAttendeeHost] : [])];
+            // In this case, the first booking user is a fixed host, so the chosen non-fixed host is added as an attendee of the booking
+            const nonFixedAttendeeHost = users.find(
+              (user) => !user.isFixed && attendeeEmailSet.has(user.email)
+            );
+            users = [...fixedHosts, ...(nonFixedAttendeeHost ? [nonFixedAttendeeHost] : [])];
+          }
         }
       }
     }
-  }
-
-  //checks what users are available
-  if (isFirstSeat) {
-    const eventTypeWithUsers: Omit<getEventTypeResponse, "users"> & {
-      users: IsFixedAwareUserWithCredentials[];
-    } = {
-      ...eventType,
-      users: users as IsFixedAwareUserWithCredentials[],
-      ...(eventType.recurringEvent && {
-        recurringEvent: {
-          ...eventType.recurringEvent,
-          count: recurringCount || eventType.recurringEvent.count,
-        },
-      }),
-    };
-    if (input.bookingData.allRecurringDates && input.bookingData.isFirstRecurringSlot) {
-      const isTeamEvent =
-        eventType.schedulingType === SchedulingType.COLLECTIVE ||
-        eventType.schedulingType === SchedulingType.ROUND_ROBIN;
-
-      const fixedUsers = isTeamEvent
-        ? eventTypeWithUsers.users.filter((user: IsFixedAwareUserWithCredentials) => user.isFixed)
-        : [];
 
-      for (
-        let i = 0;
-        i < input.bookingData.allRecurringDates.length &&
-        i < input.bookingData.numSlotsToCheckForAvailability;
-        i++
+    //checks what users are available
+    if (isFirstSeat) {
+      const eventTypeWithUsers: Omit<getEventTypeResponse, "users"> & {
+        users: IsFixedAwareUserWithCredentials[];
+      } = {
+        ...eventType,
+        users: users as IsFixedAwareUserWithCredentials[],
+        ...(eventType.recurringEvent && {
+          recurringEvent: {
+            ...eventType.recurringEvent,
+            count: recurringCount || eventType.recurringEvent.count,
+          },
+        }),
+      };
+      if (
+        bookingData.allRecurringDates &&
+        bookingData.isFirstRecurringSlot &&
+        bookingData.numSlotsToCheckForAvailability
```

Comment:
Need to add _numSlotsToCheckForAvailability_ because it could be falsy. Earlier as it had a type _any_ , it wasn't reported